open-craft/ansible-playbooks
# feat: implement vault based SSH

This PR implements Vault based SSH authentication for generic servers. We do not include this to common-server-init as we don't want everyone to have access to all servers. More info on this: https://tasks.opencraft.com/browse/SE-6299

**Test instructions**

0. Set the Vault address `export VAULT_ADDR="https://vault.opencraft.com"`
1. Log in to vault `vault login -method="github"`
2. Generate a new ssh key pair with empty passphrase for testing purposes (name should be: `se_5700`)
3. Request Vault to sign your **public** key.
    ```
    vault write -field=signed_key \
        ssh-client-signer/sign/ssh-client-signer-role \
        valid_principals="ubuntu,root" \
        public_key=@~/.ssh/se_5700.pub > ~/.ssh/se_5700-cert.pub
    ```
4. Open an SSH connection to `cert-manager`, using the command below (note that we use the **private** key here, the `-cert.pub` is automatically loaded due to naming conventions).
    ```
    ssh -i ~/.ssh/se_5700 ubuntu@cert-manager.net.opencraft.hosting
    ```
